lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -659,6 +669,41 @@ function Get-HGSContext {
     return $ctxt
 }
 
+function Get-EtcdContext {
+    Write-JujuWarning "Generating context for etcd"
+    $required = @{
+        "client_ca" = $null
```

Comment:
Is SSL always enabled in etcd? Is there a situation where you may only get non SSL info on this relation?

If SSL is optional on this relation, you need to move the certificate stuff in the ```-Optional``` option, and make sure you account for the changes bellow.

Reply:
AFAIK ssl is always enabled on the etcd charm. There is no way of telling it to not be SSL from the config